app/src/main/java/com/example/tmdb/viewmodels/DetailsViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.tmdb.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tmdb.data.MovieRepository
+import com.example.tmdb.utils.MovieItemDetail
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class DetailsViewModel(
+    private val movieRepository: MovieRepository,
+    private val movieId: Int
+): ViewModel() {
+    val movieDetailsStateFlow = MutableStateFlow(MovieItemDetail())
+
+    private suspend fun getMovieDetails(movieId: Int) {
+        movieRepository.getMovieDetails(movieId = movieId)
+            .collect { movieDetails -> movieDetailsStateFlow.emit(movieDetails) }
+    }
```

Comment:
For some reason I oversaw this, here you can also do simply
```
val movieDetailsStateFlow: StateFlow<MovieItemDetail> = movieRepository.getMovieDetails(movieId)
.stateIn(
  viewModelScope,
  SharingStarted.WhileSubscribed(),
  MovieItemDetail(),
)
```

Reply:
Still getting the same exception, however, it doesn't pop out when SharingStarted is set to Eagerly. Should I just leave it that way for now?

Reply:
Set it to Eagerly, it shouldn't make much difference since VM will live only while screen lives, so that should be the same.

Reply:
Thanks for the tips! Couldn't have done this without you.